diff -ruN libgdiplus-1.2.5./cairo/src/cairo-type1-subset.c libgdiplus-1.2.5/cairo/src/cairo-type1-subset.c
--- libgdiplus-1.2.5./cairo/src/cairo-type1-subset.c	2007-07-24 23:12:28.000000000 +0200
+++ libgdiplus-1.2.5/cairo/src/cairo-type1-subset.c	2007-08-16 01:47:48.470946851 +0200
@@ -46,6 +46,8 @@
 #include FT_OUTLINE_H
 #include FT_TYPE1_TABLES_H
 
+#include <ctype.h>
+
 typedef struct _cairo_type1_font_subset {
 
     cairo_scaled_font_subset_t *scaled_font_subset;
